### ISSUE TITLE
Fatal error: Call to undefined method stcr\wp_subscribe_reloaded::clean_email() 

### ISSUE THREAD

**Issue by Reedyseth** [stcr/subscribe-to-comments-reloaded#173]
This error was detected while trying to subscribe without commenting.

![image](https://cloud.githubusercontent.com/assets/1228711/12312390/51868ed6-ba1c-11e5-94c0-7b9459452d38.png)

**Comment by Reedyseth** [comment]
Reported here #172

**Comment by Reedyseth** [comment]
After deleting a subscription:

![image](https://cloud.githubusercontent.com/assets/1228711/12319747/a82efc60-ba61-11e5-974f-62fcb6538823.png)

**Comment by kermit4gitful** [comment]
Hi, Reedyseth. Thank you very much for making the Subscribe to Comments plugin. I can confirm that  I am also seeing both of these errors on my blog. I'm running WP 4.4.1 and using PHP 5.4.31, and the Subscribe to Comments version is 160106.

**Comment by Reedyseth** [comment]
@kermit4gitful 

This error is part of calling a function in the old code, fix is ready and will push it later in the day.

Thank you for your comment.

**Comment by jmorti** [comment]
Any update on when this fix will be released?

**Comment by Reedyseth** [comment]
It is solve on this commit https://github.com/stcr/subscribe-to-comments-reloaded/commit/203d1b949d9368724835c2d728cd6332c1e53b9a and the release is out, so you can now update.

**Comment by jmorti** [comment]
It looks like it now works, but it also looks like it deleted all the previous subscriptions. That's not good...

**Comment by jmorti** [comment]
I take that back, it didn't delete them. They are still in the database, but I dont receive notifications from the old ones and they dont show up on the manage page.

**Comment by Reedyseth** [comment]
> but it also looks like it deleted all the previous subscriptions. That's not good...

Last version(160116) did not mess with the subscription.

**Comment by jmorti** [comment]
Like I said, they are still there, but the old ones do not show up on the manage page and they are not sending out notifications. Something is still not right.

**Comment by Reedyseth** [comment]
What was your page again?

**Comment by jmorti** [comment]
Here's an example of what Im talking about:

![firefox_screenshot_2016-01-15t16-42-52 086z](https://cloud.githubusercontent.com/assets/16693709/12358845/7de9bc40-bb6c-11e5-94c6-33647abb9c8f.png)
![firefox_screenshot_2016-01-15t16-43-31 590z](https://cloud.githubusercontent.com/assets/16693709/12358846/7deffed4-bb6c-11e5-939b-0759f270cd0c.png)

**Comment by jmorti** [comment]
See, only the new subscription made after the update shows on the "manage" page. There are older ones in the database, but they are not sending notifications and I cannot manage them. Hope that helps.

**Comment by kermit4gitful** [comment]
Yay! Thank you, Reedyseth. The new version seems to be working great on our end. :)

**Comment by jmorti** [comment]
Maybe I'm confused on the manage pages. I will wait and see if the older notification still work. Thanks for all your help!

**Comment by Reedyseth** [comment]
> See, only the new subscription made after the update shows on the "manage" page. There are older ones in the database, but they are not sending notifications and I cannot manage them. Hope that helps.

I would like to debug the plugin in your Admin panel, and confirm that is an issue. Also by being there I would like to improve your notifications template, cause they are all broken. You can create a temporary user for me, use my email reedyseth at gmail dot com

**Comment by jmorti** [comment]
I had the notifications all set perfect too right before you updated. Why cant we edit the raw html anymore? I'm not sure how I feel about letting you into the backend of the site.

**Comment by Reedyseth** [comment]
> I had the notifications all set perfect too right before you updated

It is because the last update are now display in the rich editor.

> Why cant we edit the raw html anymore? I'm not sure how I feel about letting you into the backend of the site.

On that case update the text and link references on the options. See #171

**Comment by jmorti** [comment]
I changed the settings, but I still need to edit the html because I have custom styles in there. Does the editor give you the option to view the source?

**Comment by Reedyseth** [comment]
> Does the editor give you the option to view the source?

On this version no, I had many user asking a rich editor to create a HTML template wihtout touching any HTML source, perhaps in the future it will be.

**Comment by Niackery** [comment]
Hi Reedyseth,

Firstly, thanks for this plugin! A couple of issues with the last two updates. I also had custom styles in the html templates, and now they're gone, and I can't reproduce them with the rich text editor. It would be nice to have the choice... Also, with today's latest update, the changelog is gone so I have no clue what changed...

**Comment by Reedyseth** [comment]
> It would be nice to have the choice...

I had also another user reporting the same issue, in that case I would enable the source code view.

> Also, with today's latest update, the changelog is gone so I have no clue what changed...

I just fixed it, now you should be able to see it.

Thank you for taking the time to review it !

**Comment by Reedyseth** [comment]
@Niackery 

As for the Source code view, I just commit the changes. I did not go over all the update process to release a new version, I just add it to the current one so if you like, 

1. Download the latest version
2. Deactivate the plugin
3. Delete the old plugin files
4. Upload the downloaded file.
5. Activate, you should be able to see the Source code view under the tools menu.

@jmorti Source code view added, view steps above.

**Comment by Niackery** [comment]
Thanks for the changes! I have a plugin that compares the files in the repository to the files in my site and allows me to restore the originals when there are changes (Wordfence), so I just replaced the files you updated. I can see the changelog now :)

When I go to the StCR Management Page settings tab, instead of seeing a rich text editor, now I see a blank space. But if I use Inspect Element, I can see that there's a text area in a table, but the visibility is set to hidden. It looks to me like there's no js being loaded.

Also, I noticed that the Redirection plugin is showing a 404 for /wp-content/plugins/subscribe-to-comments-reloaded/includes/js/tinymce-lite/plugins/code/plugin.min.js since I replaced the files this morning, but that js file extension doesn't match any of the scripts you've registered, as far as I can tell, so that seems weird to me.

Lastly, I just noticed something about one-click unsubscribe being added to the plugin, but I didn't see anything about it in the changelog. I just want to confirm that I should go ahead and remove the StCR Better Unsubscribe plugin. Are there any settings for this new option, or does it just work automatically?

Many thanks!

**Comment by Reedyseth** [comment]
> But if I use Inspect Element, I can see that there's a text area in a table, but the visibility is set to hidden. It looks to me like there's no js being loaded.

Have you try to refresh the cache?

> I just noticed something about one-click unsubscribe being added to the plugin, but I didn't see anything about it in the changelog. I just want to confirm that I should go ahead and remove the StCR Better Unsubscribe plugin. Are there any settings for this new option, or does it just work automatically?

To enable it you have to add the tag **[oneclick_link]** in your notification content:

![image](https://cloud.githubusercontent.com/assets/1228711/12376550/5fc56362-bcb7-11e5-9bf4-5c19bdbc3d59.png)

Also you have to setup the Unsubscription page content on the same Notifications tab:

![image](https://cloud.githubusercontent.com/assets/1228711/12376556/9f6fcdfe-bcb7-11e5-8a07-c119f37e9527.png)

**Comment by Niackery** [comment]
Thanks for the info on setting up one-click unsubscribe :)

To answer your question, I cleared my caching plugin's cache, my minifying plugin's cache, and my browser cache, and that's what I saw. I just did it all again to make sure, and yep, it's all still just a blank space next to each heading. Plus, I'm getting those 404s for that URL every time I visit the StCR Management Page settings tab...

**Comment by Niackery** [comment]
Actually, I just noticed the same problem for the Notifications tab -- it's just blank space next to the headings, no text areas or rich text editors at all...

**Comment by Reedyseth** [comment]
@Niackery, what I suggest is to dissable the plugin, delete it, and then upload the latest release, download it from here:

https://downloads.wordpress.org/plugin/subscribe-to-comments-reloaded.160115.zip

**Comment by jmorti** [comment]
The link to the zip file is broken.

**Comment by Niackery** [comment]
Oops, I forgot to let you know what happened with this! I was working on my test site before, and the problem was probably that the last fix wasn't a real update and I just restored modified files. Since I hadn't updated the plugin at all on the live site yet, I figured I could just apply the update since the last fix would already be in it. It worked; there was never an issue on the live site :)

Thanks!
